### PR TITLE
UX: Redesign and refactor penalty modals

### PR DESCRIPTION
**Pull request by nbianca**
This merges the two modals code to remove duplication and implements a more consistent design.

Before:

Screenshot 2022-12-13 at 23 06 24

Screenshot 2022-12-13 at 23 06 19

After:

Screenshot 2022-12-13 at 23 06 47

Screenshot 2022-12-13 at 23 06 35

### CODE REVIEW

File: app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div class="penalty-reason-controls">
+  {{#if (eq @penaltyType "suspend")}}
+    <label class="suspend-reason-title">{{i18n "admin.user.suspend_reason_title"}}</label>
+    <ComboBox @content={{this.reasons}} @value={{this.selectedReason}} @class="suspend-reason" @onChange={{action this.setSelectedReason}} />
+
+    {{#if this.isCustomReason}}
+      <TextField @value={{this.customReason}} @class="suspend-reason" @onChange={{action this.setCustomReason}} />
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs:7]:
```suggestion
      <TextField @value={{this.customReason}} @class="suspend-reason" @onChange={{this.setCustomReason}} />
```

File: app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div class="penalty-reason-controls">
+  {{#if (eq @penaltyType "suspend")}}
+    <label class="suspend-reason-title">{{i18n "admin.user.suspend_reason_title"}}</label>
+    <ComboBox @content={{this.reasons}} @value={{this.selectedReason}} @class="suspend-reason" @onChange={{action this.setSelectedReason}} />
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs:4]:
```suggestion
    <ComboBox @content={{this.reasons}} @value={{this.selectedReason}} @class="suspend-reason" @onChange={{this.setSelectedReason}} />
```

File: app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    {{#if this.isCustomReason}}
+      <TextField @value={{this.customReason}} @class="suspend-reason" @onChange={{action this.setCustomReason}} />
+    {{/if}}
+  {{else if (eq @penaltyType "silence")}}
+    <label class="silence-reason-title">{{html-safe (i18n "admin.user.silence_reason_label")}}</label>
+    <TextField @value={{this.customReason}} @class="silence-reason" @onChange={{action this.setCustomReason}} @placeholderKey="admin.user.silence_reason_placeholder" />
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/templates/components/admin-penalty-reason.hbs:11]:
```suggestion
    <TextField @value={{this.customReason}} @class="silence-reason" @onChange={{this.setCustomReason}} @placeholderKey="admin.user.silence_reason_placeholder" />
```

File: app/assets/javascripts/discourse/tests/acceptance/admin-penalize-user-test.js [hunk 4]
```diff
@@ -150,3 +150,47 @@ acceptance("Admin - Suspend User - timeframe choosing", function (needs) {
...
+  test("shows correct timeframe options", async function (assert) {
+    await visit("/admin/users/1234/regular");
+    await click(".silence-user");
+    await click(".future-date-input-selector-header");
+
+    const options = Array.from(
+      queryAll(`ul.select-kit-collection li span.name`).map((_, x) =>
+        x.innerText.trim()
+      )
+    );
```
Comment (CvX) [inline app/assets/javascripts/discourse/tests/acceptance/admin-penalize-user-test.js:176]:
```suggestion
    const options = Array.from(queryAll(`ul.select-kit-collection li span.name`))
      .map((el) => el.innerText.trim());
```

File: app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      {{else if (eq this.penaltyType "silence")}}
+        <div class="cant-silence">{{i18n "admin.user.cant_silence"}}</div>
+      {{/if}}
+    {{/if}}
+  </ConditionalLoadingSpinner>
+</DModalBody>
+
+<div class="modal-footer">
+  <DButton @class="btn-danger perform-penalize"
+           @action={{action "penalizeUser"}}
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs:57]:
```suggestion
           @action={{this.penalizeUser}}
```

File: app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                           @label="admin.user.suspend_duration"
+                           @clearable={{false}}
+                           @input={{this.penalizeUntil}}
+                           @onChangeInput={{action (mut this.penalizeUntil)}} />
+        {{else if (eq this.penaltyType "silence")}}
+          <FutureDateInput @class="silence-until"
+                           @label="admin.user.silence_duration"
+                           @clearable={{false}}
+                           @input={{this.penalizeUntil}}
+                           @onChangeInput={{action (mut this.penalizeUntil)}} />
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs:20]: I think those two `@onChangeInput`s can be simply `{{mut this.penalizeUntil}}`

File: app/assets/javascripts/admin/addon/controllers/modals/admin-penalize-user.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    this.set("penalizeUntil", this.user?.next_penalty);
+  },
+
+  beforeClose() {
+    // prompt a confirmation if we have unsaved content
+    if (
+      !this.confirmClose &&
+      ((this.reason && this.reason.length > 1) ||
+        (this.message && this.message.length > 1))
+    ) {
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/controllers/modals/admin-penalize-user.js:60]: For a bit of readability tweak:

```suggestion
    if (this.confirmClose) {
      return;
    }

    if (
      (this.reason && this.reason.length > 1) ||
      (this.message && this.message.length > 1)
    ) {
```

File: app/assets/javascripts/admin/addon/controllers/modals/admin-penalize-user.js [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    return false;
+  },
+
+  @discourseComputed("penalizing", "penalizeUntil", "reason")
+  submitDisabled(penalizing, penalizeUntil, reason) {
+    return penalizing || isEmpty(penalizeUntil) || !reason || reason.length < 1;
+  },
+
+  @action
+  penalizeUser() {
```
Comment (CvX) [inline app/assets/javascripts/admin/addon/controllers/modals/admin-penalize-user.js:123]: Cna this be made into an async/await function? 😃

File: app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs (hunk 6 above)
Comment (nbianca) [reply app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs:20]: I tested it, but it did not work. :(

Comment (CvX) [reply app/assets/javascripts/admin/addon/templates/modal/admin-penalize-user.hbs:20]: Hm, interesting. Will take a look after it's merged!

**Pull request merged**